src/django_pg_migration_tools/operations.py

```diff
@@ -525,3 +583,256 @@ def describe(self) -> str:
+            table_name, column_name, length=10
+        )
+        return f"{table_name[:10]}_{column_name[:10]}_{suffix}"
+
+    def _is_constraint_valid(
+        self,
+        schema_editor: base_schema.BaseDatabaseSchemaEditor,
+        constraint_name: str,
+    ) -> bool:
+        cursor = schema_editor.connection.cursor()
+        cursor.execute(
+            ConstraintQueries.CHECK_CONSTRAINT_IS_VALID,
+            {"constraint_name": constraint_name},
+        )
+        return bool(cursor.fetchone())
+
+    def _alter_table_not_null(
+        self,
+        schema_editor: base_schema.BaseDatabaseSchemaEditor,
+        table_name: str,
+        column_name: str,
+    ) -> None:
+        cursor = schema_editor.connection.cursor()
+        cursor.execute(
+            psycopg_sql.SQL(NullabilityQueries.ALTER_TABLE_SET_NOT_NULL).format(
+                table_name=psycopg_sql.Identifier(table_name),
+                column_name=psycopg_sql.Identifier(column_name),
+            ),
+        )
+
+    def _alter_table_drop_not_null(
+        self,
+        schema_editor: base_schema.BaseDatabaseSchemaEditor,
+        table_name: str,
+        column_name: str,
+    ) -> None:
+        cursor = schema_editor.connection.cursor()
+        cursor.execute(
+            psycopg_sql.SQL(NullabilityQueries.ALTER_TABLE_DROP_NOT_NULL).format(
+                table_name=psycopg_sql.Identifier(table_name),
+                column_name=psycopg_sql.Identifier(column_name),
+            ),
+        )
+
+    def _alter_table_drop_constraint(
+        self,
+        schema_editor: base_schema.BaseDatabaseSchemaEditor,
+        table_name: str,
+        constraint_name: str,
+    ) -> None:
+        cursor = schema_editor.connection.cursor()
+        cursor.execute(
+            psycopg_sql.SQL(ConstraintQueries.ALTER_TABLE_DROP_CONSTRAINT).format(
+                table_name=psycopg_sql.Identifier(table_name),
+                constraint_name=psycopg_sql.Identifier(constraint_name),
+            )
+        )
+
+    def _constraint_exists(
+        self,
+        schema_editor: base_schema.BaseDatabaseSchemaEditor,
+        constraint_name: str,
+    ) -> bool:
+        cursor = schema_editor.connection.cursor()
+        cursor.execute(
+            ConstraintQueries.CHECK_EXISTING_CONSTRAINT,
+            {"constraint_name": constraint_name},
+        )
+        return bool(cursor.fetchone())
+
+    def _validate_constraint(
+        self,
+        schema_editor: base_schema.BaseDatabaseSchemaEditor,
+        table_name: str,
+        constraint_name: str,
+    ) -> None:
+        cursor = schema_editor.connection.cursor()
+        cursor.execute(
+            psycopg_sql.SQL(ConstraintQueries.ALTER_TABLE_VALIDATE_CONSTRAINT).format(
+                table_name=psycopg_sql.Identifier(table_name),
+                constraint_name=psycopg_sql.Identifier(constraint_name),
+            )
+        )
+
+    def set_null(
```

Comment:
🐼 It would be better for this "public" method to be at the top of the class, rather than after a number of "private" methods.